Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -28,6 +28,17 @@ public class DiagnosticErrorReporter : ErrorReporter {
       this.entryDocumentUri = entryDocumentUri;
     }
 
+    public IReadOnlyList<Diagnostic> GetDiagnostics(DocumentUri documentUri) {
+      rwLock.EnterReadLock();
+      try {
+        // Return a fresh list to avoid concurrency issues
```

Comment:
We could make this comment more explicit:

```cs
// Concurrency: Return a copy of the list not to expose a reference to an object that requires synchronization.
```

PS: For complete sanity, the `Diagnostic` instances would require a copy too. However, we never mutate any of these instances, so that should be fine for now (there are other locations with a  similar case). One thing I envision is to make `DafnyDocument` and its property types completely immutable.

Reply:
Done, and I added a `LATER` note about this